### PR TITLE
[release-0.15]  explicitly bring unavailable interfaces up

### PR DESCRIPTION
**Pull request by phoracek**
**What this PR does / why we need it**:

There is a bug in Kernel/NetworkManager on systems with NetworkManager
1.20, where sometimes after disconnecting a NIC from a bonding, the NIC
remains in 'unavailable' state and cannot be used for a new connection. This
is likely caused by an issue with autonegotiation where the NIC appears to
be disconnected and the only thing that can bring it available again is
explicitly calling ip link set up on it. In order to workaround
this issue until it gets solved, we iterate all devices during nmstatectl set
and if we find some with 'unavailable' we explicitly set them up.

Signed-off-by: Petr Horacek phoracek@redhat.com

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/nmstatectl/nmstatectl.go [hunk 1]
```diff
@@ -57,7 +57,12 @@ func Show(arguments ...string) (string, error) {
 }
 
 func Set(desiredState nmstatev1alpha1.State, timeout time.Duration) (string, error) {
-	return nmstatectlWithInput([]string{"set", "--no-commit", "--timeout", strconv.Itoa(int(timeout.Seconds()))}, string(desiredState.Raw))
+	var setDoneCh = make(chan struct{})
+	go setUnavailableUp(setDoneCh)
+	defer close(setDoneCh)
```
Comment (oshoval) [inline pkg/nmstatectl/nmstatectl.go]: a go routine waits for all its children ? (asking because the defer here that will close the channel)

Comment (phoracek) [reply pkg/nmstatectl/nmstatectl.go]: SetUnavailableUp will be running until we close the stopChannel which is only after the `set` is done. What is the scenario that would result in a race you think of?

File: pkg/nmstatectl/unavailable_link_workaround.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+			}
+		}
+	}, time.Second, stopCh)
+}
+
+func setLinkUp(iface string) error {
+	var stdout, stderr bytes.Buffer
+
+	cmd := exec.Command("ip", "link", "set", iface, "up")
+	cmd.Stderr = &stderr
```
[the remark this answers is not in the document]
Comment (phoracek) [reply pkg/nmstatectl/unavailable_link_workaround.go]: Maybe, this is a backport though. If there is not a functionality bug, it's not really worth changing compared to master.

If there is a better way to handle outputs, it's should be done on master in a separate PR

[the remark this answers is not in the document]
Comment (oshoval) [reply pkg/nmstatectl/unavailable_link_workaround.go]: functionality looks good, 
sounds great, thx

**Comment by qinqon** [comment]
/override pull-kubernetes-nmstate-e2e-k8s

Rebasing and tested it and it's working

**Pull request merged**